src/Diagnostics/dump_tendencies.jl

```diff
@@ -0,0 +1,239 @@
+"""
+    setup_dump_tendencies_diagnostics(
+        ::ClimateMachineConfigType,
+        interval::String,
+        out_prefix::String;
+        writer = NetCDFWriter(),
+        interpol = nothing,
+    )
+
+Create the "DumpTendencies" `DiagnosticsGroup` which contains all the
+tendencies for the simulation's `BalanceLaw`. These are collected on
+an interpolated grid (`interpol` _must_ be specified).
+
+!!! warn
+    This diagnostics group can produce a lot of output and is
+    very expensive.
+
+!!! warn
+    These diagnostics are intended for physics-debugging, and not numerics debugging, because
+    `flux` and `source` are called on interpolated data, which may not exactly match the
+    non-interpolated evaluation of fluxes and sources that the solver uses.
+"""
+function setup_dump_tendencies_diagnostics(
+    ::ClimateMachineConfigType,
+    interval::String,
+    out_prefix::String;
+    writer = NetCDFWriter(),
+    interpol = nothing,
+)
+    @assert !isnothing(interpol)
+
+    return DiagnosticsGroup(
+        "DumpTendencies",
+        Diagnostics.dump_tendencies_init,
+        Diagnostics.dump_tendencies_fini,
+        Diagnostics.dump_tendencies_collect,
+        interval,
+        out_prefix,
+        writer,
+        interpol,
+    )
+end
+
+# helpers for PV and tendency names
+params(::T) where {T} = T.parameters
+unval(::Val{i}) where {i} = i
+param_suffix(p, ::Val{n}) where {n} = string("_", unval(p[1]))
+param_suffix(p, ::Val{0}) = ""
+param_suffix(p) = param_suffix(p, Val(length(p)))
+prog_name(pv::PV) where {PV} = string(nameof(PV), param_suffix(params(pv)))
+tend_name(t) = String(nameof(typeof(t)))
+
+function precompute_args(
+    bl,
+    state,
+    aux,
+    diffusive,
+    hyperdiffusive,
+    t,
+    direction,
+)
+    _args_fx1 = (; state, aux, t, direction)
+    _args_fx2 = (; state, aux, t, diffusive, hyperdiffusive)
+    _args_src = (; state, aux, t, direction, diffusive)
+
+    cache_fx1 = precompute(bl, _args_fx1, Flux{FirstOrder}())
+    cache_fx2 = precompute(bl, _args_fx2, Flux{SecondOrder}())
+    cache_src = precompute(bl, _args_src, Source())
```

Comment:
Another thing that would be super useful would be to return/flatten this named tuple and dump into the .nc file. I think we can tackle that in a separate PR, and we'll need to filter out some keys (e.g., `ts::ThermodynamicState`).